Updated docstring for get_cpds method in BayesianNetwork

B3CODER opened the pull request:
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

 Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
 Make sure you are making a pull request against the dev branch (left side). Also you should start your branch off our dev.
 Check the commit's or even all commits' message styles matches our requested structure.

- Fixes #933

### List of changes to the codebase in this pull request
- Updated the docstring for the get_cpds method in BayesianNetwork to clarify the method's behavior when the node argument is specified and when it is not.
- Added more detailed descriptions to improve the documentation's clarity and usefulness for users.

B3CODER committed:
Updated docstring for get_cpds method in BayesianNetwork

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,8 +279,17 @@ def get_cpds(self, node=None):
         -------
-        A list of TabularCPDs: list
-
+        - If 'node' is specified: 
```
I would keep the first line in the form of `name : type` form. Have a look at the numpydoc style for details: https://numpydoc.readthedocs.io/en/latest/format.html#returns

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,8 +279,17 @@ def get_cpds(self, node=None):
+            
         Examples
         --------
         >>> from pgmpy.models import BayesianNetwork
```
I would completely switch out this example with something similar to the `get_example_model` example I had added to the issue.

ankurankan reviewed (changes requested):
@B3CODER Thanks for creating the PR. I have added a few small comments to the changes.

B3CODER committed:
Updated get_cpds function docstring to proper Returns format in BayesianNetwork

B3CODER committed:
Formatted BayesianNetwork.py using black

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -291,6 +299,13 @@ def get_cpds(self, node=None):
         ...                  ['intel', 'diff'], [2, 2])
         >>> student.add_cpds(cpd)
         >>> student.get_cpds()
+
```
Could you please update the examples here such that we use one of the example models? If you search for `get_example_model`, you will find plenty of docstring examples using the `get_example_model` method to create a model.

B3CODER committed:
Updated examples in BayesianNetwork.py based on review feedback

B3CODER commented:
### Changes Made:
- Updated the `get_cpds` method to provide consistent output for both the list of all CPDs and the CPD of a specific node.
- Added error handling for nodes not present in the model.

### Screenshot of the modified code:
![image](https://github.com/user-attachments/assets/518ca3a2-3e7e-4181-ba02-cb40f088208f)

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,18 +279,35 @@ def get_cpds(self, node=None):
-        >>> student.add_cpds(cpd)
-        >>> student.get_cpds()
+        >>> import pprint
+        >>> random_model = BayesianNetwork.get_random(n_nodes=8, edge_prob=0.4, n_states=2, latents=True)
```
@B3CODER Here, you are currently using the `get_random` method to generate the model. I was suggesting to use the `get_example_model` method to generate the model. Something like this:
```python
from pgmpy.utils import get_example_model
model = get_example_model('cancer')
```
And then show examples similar to the ones that you have shown here.

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -608,7 +625,7 @@ def fit_update(self, data, n_prev_samples=None, n_jobs=1):
         >>> from pgmpy.utils import get_example_model
         >>> from pgmpy.sampling import BayesianModelSampling
-        >>> model = get_example_model('alarm')
+        >>> model = ('alarm')
```
You would need to revert this.

ankurankan commented on pgmpy/utils/utils.py:
```diff
@@ -63,7 +63,7 @@ def get_example_model(model):
     -------
     >>> from pgmpy.data import get_example_model
-    >>> model = get_example_model(model='asia')
+    >>> model = (model='asia')
```
This as well.

ankurankan commented:
@B3CODER 

> Updated the get_cpds method to provide consistent output for both the list of all CPDs and the CPD of a specific node.
> Added error handling for nodes not present in the model.

I do not see any changes related to these in the PR. Have you pushed your local changes? Also, I would not make those changes as it would break a lot of other parts of the codebase.

B3CODER committed:
Updated get_cpds method: ensured consistent return type and improved error handling

B3CODER committed:
Updated get_cpds method: ensured consistent return type and improved error handling

B3CODER committed:
Updated get_cpds method: ensured consistent return type and improved error handling

B3CODER committed:
Fix: Applied changes based on pre-commit checks

ankurankan commented:
@B3CODER The code formatting changes were needed only for the files that you had changes in, not for the whole repo. You can revert these by running:
```
git revert d31ba54
```

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
-        >>> student.get_cpds()
+        >>> from pgmpy.utils import get_example_model
+        >>> model = get_example_model('asia')
+        >>> cpds=model.get_cpds()
```
Space before and after `=`. Should be: `cpds = model.get_cpds()`

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
+        <TabularCPD representing P(smoke:2) at 0x7dbbd9bd8f70>,
+        <TabularCPD representing P(tub:2 | asia:2) at 0x7dbbd9bda860>,
+        <TabularCPD representing P(xray:2 | either:2) at 0x7dbbd9bd9a80>]
+        >>>  cpds = model.get_cpds('bronc')     # Returns the CPD of node
```
Change cpds to cpd as we expect the method to only return a single cpd

ankurankan commented on pgmpy/models/BayesianNetwork.py:
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
+        >>>  cpds = model.get_cpds('bronc')     # Returns the CPD of node
+        >>>  cpds
+        <TabularCPD representing P(bronc:2 | smoke:2) at 0x7dbbd9bda3e0>
+        >>> cpds = model.get_cpds('tbu')
```
You can remove this example with the error.

ankurankan reviewed (changes requested):
I have left a few comments, everything else looks good.

ankurankan commented:
@B3CODER This PR can be closed. Right?

B3CODER commented:
Yes sir

On Wed, 16 Oct 2024, 2:43 pm Ankur Ankan, ***@***.***> wrote:

> @B3CODER <https://github.com/B3CODER> This PR can be closed. Right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pgmpy/pgmpy/pull/1835#issuecomment-2416201944>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVZA3LWF557PYPV3FABGKBLZ3YU3BAVCNFSM6AAAAABP3I7ETOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJWGIYDCOJUGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

B3CODER commented:
Sir, any issue regarding PR?

On Wed, 16 Oct 2024 at 14:49, BHAVYA SHAH ***@***.***> wrote:

> Yes sir
>
> On Wed, 16 Oct 2024, 2:43 pm Ankur Ankan, ***@***.***>
> wrote:
>
>> @B3CODER <https://github.com/B3CODER> This PR can be closed. Right?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/pgmpy/pgmpy/pull/1835#issuecomment-2416201944>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AVZA3LWF557PYPV3FABGKBLZ3YU3BAVCNFSM6AAAAABP3I7ETOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJWGIYDCOJUGQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
